src/useForm.tsx

```diff
@@ -100,22 +101,24 @@ function createFields<T>(
             options?: SetValueOptions
           ): SetValueReturnType<U> => {
             const isValid = new Promise<boolean>(resolve => {
-              setState(currentState =>
-                produce(currentState, updatedState => {
+              setState(currentState => {
+                const fieldValues = extractValuesFromFieldsState(currentState)
+                return produce(currentState, updatedState => {
                   set(updatedState, [...path, 'value'], updatedValue)
                   set(updatedState, [...path, 'touched'], true)
 
                   if (options?.runValidation) {
                     const error = maybeGetFirstValidationError(
                       updatedValue,
-                      currentState,
```

Comment:
Here, we really want the field values (`T`) rather than the `FieldsState<T>`(has extra properties on it) so I made a tweak here by hoisting the call to `getValue` (now called `extractValuesFromFieldsState`) up here.